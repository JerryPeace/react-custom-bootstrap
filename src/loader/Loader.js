import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';
import CustomPropTypes from '../utils/CustomPropTypes';
import styleMaps from '../styleMaps';

class Loader extends React.Component {

    static propTypes = {
        loaderText: PropTypes.string,
        bsSize: CustomPropTypes.keyOf(styleMaps.SIZES),
        bsClass: PropTypes.string
    };

    static defaultProps = {
        bsClass: 'tm-loader',
        bsSize: 'sm'
    };

    renderContent() {
        const {loaderText, children} = this.props;
        if (loaderText || children) {
            return (
                <span className={styleMaps.CLASSES['loader-content']}>
                    {loaderText}
                    {children}
                </span>
            );
        } else {
            return null;
        }
    }

    render() {
        const {className, bsClass, bsSize, loaderText, children, ...props} = this.props;
        const bsCls = bootstrapUtils.getCls(this.props);
        return (
            <span className={classNames('loader', className)} {...props}>
                <span className={classNames(bsCls)} />
                {this.renderContent()}
            </span>
        );
    }
}

export default Loader;
