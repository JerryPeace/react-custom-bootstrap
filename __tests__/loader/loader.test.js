import React from 'react';
import { mount } from 'enzyme';
import Loader from 'loader/Loader';
import toJSON from'enzyme-to-json';

describe('Loader', () => {

    let loader, inst;

    describe('Default Loader', () => {

        beforeEach(() => {
            loader = mount(<Loader />);
        });

        test('should render corrently', () => {
            expect(toJSON(loader)).toMatchSnapshot();
        });

    });

    describe('Loader Text', () => {

        beforeEach(() => {
            loader = mount(<Loader loaderText="Loading"/>);
        });

        test('should render corrently', () => {
            expect(toJSON(loader)).toMatchSnapshot();
        });

    });

    describe('Children Elmnt', () => {

        const childrenElmnt = <p>Loading...</p>

        beforeEach(() => {
            loader = mount(<Loader children={childrenElmnt}/>);
        });

        test('should render corrently', () => {
            expect(toJSON(loader)).toMatchSnapshot();
        });

    });
});
