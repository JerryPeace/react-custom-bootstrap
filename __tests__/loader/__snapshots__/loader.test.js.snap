// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Children Elmnt should render corrently 1`] = `
<Loader
  bsClass="tm-loader"
  bsSize="sm"
>
  <span
    className="loader"
  >
    <span
      className="tm-loader tm-loader-sm"
    />
    <span
      className="loader-content"
    >
      <p>
        Loading...
      </p>
    </span>
  </span>
</Loader>
`;

exports[`Loader Default Loader should render corrently 1`] = `
<Loader
  bsClass="tm-loader"
  bsSize="sm"
>
  <span
    className="loader"
  >
    <span
      className="tm-loader tm-loader-sm"
    />
  </span>
</Loader>
`;

exports[`Loader Loader Text should render corrently 1`] = `
<Loader
  bsClass="tm-loader"
  bsSize="sm"
  loaderText="Loading"
>
  <span
    className="loader"
  >
    <span
      className="tm-loader tm-loader-sm"
    />
    <span
      className="loader-content"
    >
      Loading
    </span>
  </span>
</Loader>
`;
